app/utils/graphql.ts

```diff
@@ -24,6 +24,10 @@ const DEFAULT_SWR_OPTIONS: SWRConfiguration = {
    * Dont refetch on network recon
    */
   revalidateOnReconnect: false,
+  /**
+   * keep the previoius data while refetching
+   */
+  keepPreviousData: true,
```

Comment:
turns out there was already an SWR option that does what `useDataCache` did:
https://swr.vercel.app/docs/advanced/understanding.en-US#return-previous-data-for-better-ux

👍